packages/order-utils/src/signature_utils.ts

```diff
@@ -192,36 +194,83 @@ export const signatureUtils = {
         }
     },
     /**
-     * Signs an orderHash and returns it's elliptic curve signature and signature type.
-     * This method currently supports TestRPC, Geth and Parity above and below V1.6.6
-     * @param   orderHash       Hex encoded orderHash to sign.
+     * Signs an order and returns its elliptic curve signature and signature type. First `eth_signTypedData` is requested
+     * then a fallback to `eth_sign` if not available on this provider.
+     * @param   order The Order to sign.
      * @param   signerAddress   The hex encoded Ethereum address you wish to sign it with. This address
      *          must be available via the Provider supplied to 0x.js.
-     * @param   signerType Different signers add/require different prefixes to be prepended to the message being signed.
-     *          Since we cannot know ahead of time which signer you are using, you must supply a SignerType.
      * @return  A hex encoded string containing the Elliptic curve signature generated by signing the orderHash and the Signature Type.
      */
-    async ecSignOrderHashAsync(
-        provider: Provider,
-        orderHash: string,
-        signerAddress: string,
-        signerType: SignerType,
-    ): Promise<string> {
+    async ecSignOrderAsync(provider: Provider, order: Order, signerAddress: string): Promise<string> {
+        try {
+            const signatureHex = signatureUtils.ecSignTypedDataOrderAsync(provider, order, signerAddress);
+            return signatureHex;
+        } catch (err) {
```

Comment:
We can check for our provider errors but a third party provider may thrown an error we don't map (like an Ethers provider). What were you thinking we could do with checking a specific error?

Reply:
Ok, then let's put a `// HACK: ` comment here and explain this so the next dev understands it. Can you think of any errors it would catch that are definitively not related to the endpoint not existing? Perhaps we could at least exclude some we know we should ignore.